src/main/java/org/broadinstitute/hellbender/utils/pileup/PileupElement.java

```diff
@@ -21,13 +26,27 @@
     public static final byte T_FOLLOWED_BY_INSERTION_BASE = (byte) 89;
     public static final byte G_FOLLOWED_BY_INSERTION_BASE = (byte) 90;
 
-    protected final GATKRead read;         // the read this base belongs to
-    protected final int offset;                 // the offset in the bases array for this base
+    final GATKRead read;         // the read this base belongs to
+    final int offset;            // the offset in the bases array for this base
 
     private final CigarElement currentCigarElement;
     private final int currentCigarOffset;
     private final int offsetInCurrentCigar;
 
+    public static final Comparator<PileupElement> COMPARATOR = (p1, p2) -> {
```

Comment:
This `Comparator` seems strange/wrong to me -- what if p1's offset is greater than p2's offset, but p1's start position is much less than p2's start position? Then this comparator would say that p1 is greater than p2 (because p1.offset > p2.offset), even though p1's base is well before p2's base on the genome.